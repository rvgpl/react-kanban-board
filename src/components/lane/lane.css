.lane {
  background: #e3e8ea;
}
.lane-title {
  background-color: var(--primaryBgColor);
  color: var(--primaryColor);
  line-height: 1;
  font-size: 1.4rem;
  font-weight: bold;
  padding: 1rem;
}

.lane-items {
  padding: 1rem;
  border-top: 0;
  display: flex;
  flex-direction: column;
  gap: 1rem;
}