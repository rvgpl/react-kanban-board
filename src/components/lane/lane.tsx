import React from "react";
import "./lane.css";

type ILaneProps = {
  title: string;
};
const Lane: React.FC<ILaneProps> = ({ title, children }) => {
  return (
    <section className="lane">
      <h2 className="lane-title">{title}</h2>
      <div className="lane-items">{children}</div>
    </section>
  );
};

export { Lane };
