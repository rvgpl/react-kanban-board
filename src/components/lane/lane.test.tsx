import { render } from "@testing-library/react";
import { Lane } from "./lane";

describe("Lane", () => {
  it("should render", () => {
    const { container } = render(
      <Lane title="todo">
        <p>Item 1</p>
        <p>Item 2</p>
      </Lane>
    );
    expect(container).toMatchSnapshot();
  });
});
