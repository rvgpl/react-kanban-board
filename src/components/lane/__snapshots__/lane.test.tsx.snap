// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lane should render 1`] = `
<div>
  <section
    class="lane"
  >
    <h2
      class="lane-title"
    >
      todo
    </h2>
    <div
      class="lane-items"
    >
      <p>
        Item 1
      </p>
      <p>
        Item 2
      </p>
    </div>
  </section>
</div>
`;
